src/main/java/com/coveo/spillway/storage/RedisStorage.java

```diff
@@ -69,6 +69,12 @@
 
   private static final String KEY_SEPARATOR_SUBSTITUTE = "_";
   private static final String WILD_CARD_OPERATOR = "*";
+  private static final String COUNTER_SCRIPT =
+      "local counter = redis.call('INCRBY', KEYS[1], ARGV[1]); "
+          + "if counter  > tonumber(ARGV[2]) + tonumber(ARGV[1])"
+          + "then counter = redis.call('INCRBY', KEYS[1], -ARGV[1]) "
```

Comment:
is there a reason to increase then revert, as opposed to checking first?

Reply:
It was just to avoid checks that needed to be done when you read the value first time. The GET function returns `nil` if the value does not exist as opposed to `incryby` where it returns 1 if not found. Also GET returns a string value that requires extra tonumber casting etc.